test/clojure/core_test/atom.cljc

```diff
@@ -0,0 +1,142 @@
+        (is (thrown? #?(:cljs :default :clj Exception :cljr Exception)
+                     (atom {} :validator (constantly false))))
+        (is (thrown? #?(:cljs :default :clj Exception :cljr Exception)
+                     (atom {} :validator (constantly (throw (ex-info "boom" {})))))))
+
+      (testing "conditional validators are obeyed at creation, swap! and reset!"
+        (is (thrown? #?(:cljs :default :clj Exception :cljr Exception)
+                     (atom #{} :validator (fn [v] (some string? v)))))
+        (let [some-strings (atom #{"str"} :validator (fn [v] (some string? v)))]
+          (is (= #{"str" :not-a-string} (swap! some-strings conj :not-a-string)))
+          (is (thrown? #?(:cljs :default :clj Exception :cljr Exception)
+                       (swap! some-strings disj "str")))
+          (is (= #{"str"} (swap! some-strings disj :not-a-string)))
+          (is (thrown? #?(:cljs :default :clj Exception :cljr Exception)
+                       (reset! some-strings #{})))
+          (is (thrown? #?(:cljs :default :clj Exception :cljr Exception)
+                       (reset! some-strings :neither-string-nor-set)))
+          (is (= #{"str"} (deref some-strings)))
+          (is (= #{"some other string"} (reset! some-strings #{"some other string"})))
+          (is (= #{"some other string"} (deref some-strings))))
+        
+        (let [all-strings (atom #{} :validator (fn [v] (every? string? v)))]
+          (is (= #{"str"} (swap! all-strings conj "str")))
+          (is (= #{} (swap! all-strings disj "str")))
+          (is (thrown? #?(:cljs :default :clj Exception :cljr Exception)
+                       (reset! all-strings :neither-string-nor-set)))
+          (is (thrown? #?(:cljs :default :clj Exception :cljr Exception)
+                       (reset! all-strings #{:not-a-string})))
+          (is (= #{"new string"} (reset! all-strings #{"new string"})))
+          (is (= #{"new string"} (deref all-strings))))))
+
+    (testing "metadata and validator function together"
+      (let [m {:foo "foo"}
+            vf even?
+            the-atom (atom 0 :validator vf :meta m)
+            the-atom2 (atom 0 :meta m :validator vf)]
+        (is (= vf (get-validator the-atom)))
+        (is (= m (meta the-atom)))
+        (is (= vf (get-validator the-atom2)))
+        (is (= m (meta the-atom2)))))
+
+    (testing "atom accepts all values"
+      (are [v] (let [the-atom (atom v)]
+                 (and (= v (deref the-atom))
+                      #?(:cljs (is (satisfies? cljs.core/IAtom the-atom))
+                         :clj (is (instance? clojure.lang.Atom the-atom)))))
+        'sym
+        `sym
+        "string"
+        1
+        1.0
+        #?(:clj 111/7) ; ratios not supported in cljs
+        \newline
+        nil
+        true
+        false
+        ##Inf
+        :kw
+        :ns/kw
+        '(one two three)
+        [1 2 3]
+        {:k "value"}
```

Comment:
I would add a regex literal here for completeness

Reply:
I think we can pass on this, due to the oddities of regex equality and the unrelatedness to `atom` in particular.